starter-code/ShoppingListWithDetailView/app/src/main/java/ly/generalassemb/drewmahrt/shoppinglistdetailview/MainActivity.java

```diff
@@ -20,16 +21,21 @@ protected void onCreate(Bundle savedInstanceState) {
         DBAssetHelper dbSetup = new DBAssetHelper(MainActivity.this);
         dbSetup.getReadableDatabase();
 
-        //Setup the RecyclerView
-        RecyclerView shoppingListRecyclerView = (RecyclerView) findViewById(R.id.shopping_list_recyclerview);
+        Fragment listFragment = ListFragment.newInstance(null,this);
 
-        ShoppingSQLiteOpenHelper db = ShoppingSQLiteOpenHelper.getInstance(this);
-        List<ShoppingItem> shoppingList = db.getShoppingList();
+        getSupportFragmentManager().beginTransaction().add(R.id.fragment_container,listFragment).commit();
 
-        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(this,LinearLayoutManager.VERTICAL,false);
+    }
 
-        shoppingListRecyclerView.setLayoutManager(linearLayoutManager);
-        shoppingListRecyclerView.setAdapter(new ShoppingListAdapter(shoppingList));
+    @Override
+    public void OnClick(ShoppingItem item) {
+        Bundle bundle = new Bundle();
+        bundle.putString("name",item.getName());
+        bundle.putString("description",item.getDescription());
+        bundle.putString("price",item.getPrice());
+        bundle.putString("type",item.getType());
```

Comment:
One of the requirements was "Gather the details for the selected item from the database (not passed through the intent)". Technically, you are not passing via an intent, but pretty much the same thing. This works OK in this case, but what if the item had 100 fields of data rather than 4? You'd want to pass just the ID value, then query the database again from the detail fragment to get the one item you need.